### PR TITLE
[Request][8.15 & Serverless] Alert suppression for ES|QL and ML rules 

### PR DESCRIPTION
**Pull request by nastasha-solomon**
Fixes https://github.com/elastic/security-docs/issues/5516 and https://github.com/elastic/security-docs/issues/5517

**ESS previews**
- Alert suppression for ES|QL rules: 
  - [Create an ES|QL rule](https://security-docs_bk_5568.docs-preview.app.elstc.co/guide/en/security/master/rules-ui-create.html#create-esql-rule): Added an optional step for configuring alert suppression
  - [ES|QL query design considerations](https://security-docs_bk_5568.docs-preview.app.elstc.co/guide/en/security/master/rules-ui-create.html#esql-query-design): Added a bullet about configuring suppression on non-aggregating queries 
- Alert suppression for ML rules: 
  - [Create a machine learning rule](https://security-docs_bk_5568.docs-preview.app.elstc.co/guide/en/security/master/rules-ui-create.html#create-ml-rule): Added an optional step for configuring alert suppression. 
- [Main docs for alert suppression](https://security-docs_bk_5568.docs-preview.app.elstc.co/guide/en/security/master/alert-suppression.html): Minor updates to show that alert suppression is supported for ES|QL and ML rules 
- API docs
  - [Create rule](https://security-docs_bk_5568.docs-preview.app.elstc.co/guide/en/security/master/rules-api-create.html#opt-fields-alert-suppression-create)
   - [Update rule](https://security-docs_bk_5568.docs-preview.app.elstc.co/guide/en/security/master/rules-api-update.html#opt-fields-alert-suppression-update)

**Serverless preview** 
(Same updates as ESS docs, minus API doc updates)
- Alert suppression for ES|QL rules: 
  - [Create an ES|QL rule](https://elastic-dot-co-docs-production-q74rgn131-elastic-dev.vercel.app/current/serverless/security/rules-create#create-an-esql-rule)
  - [ES|QL query design considerations](https://elastic-dot-co-docs-production-q74rgn131-elastic-dev.vercel.app/current/serverless/security/rules-create#query-design-considerations) 
- [Alert suppression for ML rules](https://elastic-dot-co-docs-production-q74rgn131-elastic-dev.vercel.app/current/serverless/security/rules-create#create-a-machine-learning-rule)
- [Main docs for alert suppression](https://elastic-dot-co-docs-production-q74rgn131-elastic-dev.vercel.app/current/serverless/security/alert-suppression)

### CODE REVIEW

**Comment by elasticdocs** [comment]
:rocket: Built _elastic-dot-co-docs-preview-docs_ successfully!

* https://elastic-dot-co-docs-production-lm9z7w87j-elastic-dev.vercel.app

* Built with commit 943690e1462ac2dca1a8956c0ea89539bd859ea4

> Issues? Visit #next-docs in Slack

File: docs/detections/rules-ui-create.asciidoc [hunk 1]
```diff
@@ -466,6 +471,8 @@ If the `LIMIT` value and **Max alerts per run** value are different, the rule us
 
 - When writing an aggregating query, use the {ref}/esql-commands.html#esql-stats-by[`STATS...BY`] command with fields that you want to search and filter for after alerts are created. For example, using the `host.name`, `user.name`, `process.name` fields with the `BY` operator of the `STATS...BY` command returns these fields in alert documents, and allows you to search and filter for them from the Alerts table. 
 
+- When configuring alert suppression on a non-aggregating query, avoid sorting alerts by descending `@timestamp` order. Doing so can impact the number of alerts that are suppressed, especially if the number of alerts generated is higher than the maximum number of alerts the rule can create when it runs.
```
Comment (vitaliidm) [inline docs/detections/rules-ui-create.asciidoc:474]: I would rephrase:
 ```suggestion
- When configuring alert suppression on a non-aggregating query, it is recommended to sort results by ascending `@timestamp` order. It will ensure the number of alerts will be suppressed, especially if the number of alerts generated is higher than the maximum number of alerts the rule can create when it runs.
```

File: docs/serverless/alerts/alert-suppression.mdx [hunk 2]
```diff
@@ -31,21 +33,18 @@ Alert suppression is not available for Elastic prebuilt rules. However, if you w
...
-    * **Indicator match, event correlation (non-sequence queries only), or new terms rule** - Alerts with the specified field name and identical array values are grouped together. For example, if you suppress alerts by `destination.ip` of `[127.0.0.1, 127.0.0.2, 127.0.0.3]`, alerts with the entire array are grouped and only one alert is created for the group. 
+    * **Custom query or threshold rules:** Alerts are grouped by each unique value. For example, if you suppress alerts by `destination.ip` of `[127.0.0.1, 127.0.0.2, 127.0.0.3]`, alerts will be suppressed separately for each value of `127.0.0.1`, `127.0.0.2`, and `127.0.0.3`. 
+    * **Indicator match, event correlation (non-sequence queries only), new terms, or ((esql)) rule:** Alerts with the specified field name and identical array values are grouped together. For example, if you suppress alerts by `destination.ip` of `[127.0.0.1, 127.0.0.2, 127.0.0.3]`, alerts with the entire array are grouped and only one alert is created for the group.
```
Comment (vitaliidm) [inline docs/serverless/alerts/alert-suppression.mdx:47]: ML rule here?

File: docs/detections/alert-suppression.asciidoc [hunk 3]
```diff
@@ -25,25 +27,22 @@ NOTE: Alert suppression is not available for Elastic prebuilt rules. However, if
...
-* **Indicator match rule, event correlation (non-sequence queries only), or new terms rule** - Alerts with the specified field name and identical array values are grouped together. For example, if you suppress alerts by `destination.ip` of `[127.0.0.1, 127.0.0.2, 127.0.0.3]`, alerts with the entire array are grouped and only one alert is created for the group.
+* **Custom query or threshold rules:** A group of alerts is created for each value. For example, if you suppress alerts by `destination.ip` of `[127.0.0.1, 127.0.0.2, 127.0.0.3]`, alerts will be suppressed separately for each value of `127.0.0.1`, `127.0.0.2`, and `127.0.0.3`. 
+* **Indicator match rule, event correlation (non-sequence queries only), new terms, or {esql} rule:** Alerts with the specified field name and identical array values are grouped together. For example, if you suppress alerts by `destination.ip` of `[127.0.0.1, 127.0.0.2, 127.0.0.3]`, alerts with the entire array are grouped and only one alert is created for the group.
```
[the remark this answers is not in the document]
Comment (nastasha-solomon) [reply docs/detections/alert-suppression.asciidoc]: Resolved in [893ab42](https://github.com/elastic/security-docs/pull/5568/commits/893ab42f59ed03a3488dc16698d940a137cdfda0)

File: docs/detections/rules-ui-create.asciidoc (hunk 1 above)
Comment (nastasha-solomon) [reply docs/detections/rules-ui-create.asciidoc]: Updated in [893ab42](https://github.com/elastic/security-docs/pull/5568/commits/893ab42f59ed03a3488dc16698d940a137cdfda0)

File: docs/serverless/alerts/alert-suppression.mdx (hunk 2 above)
Comment (nastasha-solomon) [reply docs/serverless/alerts/alert-suppression.mdx]: Same comments as above

**Review by vitaliidm — DISMISSED** [review]

File: docs/detections/rules-ui-create.asciidoc [hunk 4]
```diff
@@ -46,6 +46,10 @@ then select:
 NOTE: If a required job isn't currently running, it will automatically start when you finish configuring and enable the rule.
 .. The anomaly score threshold above which alerts are created.
 +
+. preview:[] (Optional, https://www.elastic.co/pricing[Platinum or higher subscription] required) Use *Suppress alerts by* to reduce the number of repeated or duplicate alerts created by the rule. Refer to <<alert-suppression>> for more information.
++
+NOTE: Because {ml} rules generate alerts from anomalies, alert documents don't contain source event fields and you can only use anomaly fields when configuring alert suppression.
```
Comment (rylnd) [inline docs/detections/rules-ui-create.asciidoc:51]: I tried simplifying the phrasing a bit for clarity:

```suggestion
NOTE: Because {ml} rules generate alerts from anomalies, which don't contain source event fields, you can only use anomaly fields when configuring alert suppression.
```

**Review by rylnd — APPROVED** [review]
This looks good to me! I offered one suggestion about the phrasing on an ML explanation, but I think it's clear as is, too.

File: docs/serverless/rules/rules-ui-create.mdx [hunk 5]
```diff
@@ -49,6 +49,12 @@ To create or edit ((ml)) rules, you need an appropriate user role. Additionally,
 
     1. The anomaly score threshold above which alerts are created.
 
+1. <DocBadge template="technical preview" /> (Optional) Use **Suppress alerts by** to reduce the number of repeated or duplicate alerts created by the rule. Refer to <DocLink slug="/serverless/security/alert-suppression">Suppress detection alerts</DocLink> for more information.
+
+    <DocCallOut title="Note">
+    Because ((ml)) rules generate alerts from anomalies, alert documents don't contain source event fields and you can only use anomaly fields when configuring alert suppression.
```
Comment (nastasha-solomon) [inline docs/serverless/rules/rules-ui-create.mdx:55]:
```suggestion
    Because ((ml)) rules generate alerts from anomalies, which don't contain source event fields, you can only use anomaly fields when configuring alert suppression.
```

File: docs/detections/alert-suppression.asciidoc [hunk 6]
```diff
@@ -25,31 +27,28 @@ NOTE: Alert suppression is not available for Elastic prebuilt rules. However, if
...
-* **Indicator match rule, event correlation (non-sequence queries only), or new terms rule** - Alerts with the specified field name and identical array values are grouped together. For example, if you suppress alerts by `destination.ip` of `[127.0.0.1, 127.0.0.2, 127.0.0.3]`, alerts with the entire array are grouped and only one alert is created for the group.
+* **Custom query or threshold rules:** A group of alerts is created for each value. For example, if you suppress alerts by `destination.ip` of `[127.0.0.1, 127.0.0.2, 127.0.0.3]`, alerts will be suppressed separately for each value of `127.0.0.1`, `127.0.0.2`, and `127.0.0.3`. 
+* **Indicator match rule, event correlation (non-sequence queries only), new terms, {esql}, or {ml} rule:** Alerts with the specified field name and identical array values are grouped together. For example, if you suppress alerts by `destination.ip` of `[127.0.0.1, 127.0.0.2, 127.0.0.3]`, alerts with the entire array are grouped and only one alert is created for the group.
```
Comment (natasha-moore-elastic) [inline docs/detections/alert-suppression.asciidoc:45]:
```suggestion
* **Indicator match, event correlation (non-sequence queries only), new terms, {esql}, or {ml} rules:** Alerts with the specified field name and identical array values are grouped together. For example, if you suppress alerts by `destination.ip` of `[127.0.0.1, 127.0.0.2, 127.0.0.3]`, alerts with the entire array are grouped and only one alert is created for the group.
```

**Review by natasha-moore-elastic — DISMISSED** [review]
One nit suggestion, otherwise looks good!

File: docs/serverless/alerts/alert-suppression.mdx [hunk 7]
```diff
@@ -31,28 +33,25 @@ Alert suppression is not available for Elastic prebuilt rules. However, if you w
...
-    * **Indicator match, event correlation (non-sequence queries only), or new terms rule** - Alerts with the specified field name and identical array values are grouped together. For example, if you suppress alerts by `destination.ip` of `[127.0.0.1, 127.0.0.2, 127.0.0.3]`, alerts with the entire array are grouped and only one alert is created for the group. 
+    * **Custom query or threshold rules:** Alerts are grouped by each unique value. For example, if you suppress alerts by `destination.ip` of `[127.0.0.1, 127.0.0.2, 127.0.0.3]`, alerts will be suppressed separately for each value of `127.0.0.1`, `127.0.0.2`, and `127.0.0.3`. 
+    * **Indicator match, event correlation (non-sequence queries only), new terms, ((esql)), or ((ml)) rule:** Alerts with the specified field name and identical array values are grouped together. For example, if you suppress alerts by `destination.ip` of `[127.0.0.1, 127.0.0.2, 127.0.0.3]`, alerts with the entire array are grouped and only one alert is created for the group.
```
Comment (nastasha-solomon) [inline docs/serverless/alerts/alert-suppression.mdx:47]:
```suggestion
    * **Indicator match, event correlation (non-sequence queries only), new terms, ((esql)), or ((ml)) rules:** Alerts with the specified field name and identical array values are grouped together. For example, if you suppress alerts by `destination.ip` of `[127.0.0.1, 127.0.0.2, 127.0.0.3]`, alerts with the entire array are grouped and only one alert is created for the group. 
```

**Comment by nastasha-solomon** [comment]
Tentatively updating this PR's publication date to next Tuesday (July 30) since the [PR](https://github.com/elastic/security-docs/pull/5568) that enables the feature flag by default in `main` might be merged this week.

cc: @vitaliidm @rylnd

**Review by natasha-moore-elastic — APPROVED** [review]

**Comment by nastasha-solomon** [comment]
Verified that suppression is available in the UI for Serverless (huzzah!), so merging this PR now!
